scripts-dev/complement.sh

```diff
@@ -1,22 +1,40 @@
-#! /bin/bash -eu
+#!/usr/bin/env bash
 # This script is designed for developers who want to test their code
 # against Complement.
 #
 # It makes a Synapse image which represents the current checkout,
-# then downloads Complement and runs it with that image.
+# builds a synapse-complement image on top, then runs tests with it.
+#
+# By default the script assumes that a Complement checkout exists next to
+# the current Synapse checkout directory. This can be overridden by setting
+# the COMPLEMENT_DIR env var to the path to your Complement checkout.
+#
+# A regular expression of test method names can be supplied as the first
+# argument to the script. Complement will then only run those tests. If
+# no regex is supplied, all tests are run. Ex.
+#
+# ./complement.sh "TestOutboundFederation(Profile|Send)"
+#
+
+# Exit if a line returns a non-zero exit code
+set -e
+
+COMPLEMENT_DIR="${COMPLEMENT_DIR:-$(dirname $0)/../../complement}"
```

Comment:
I'm not totally comfortable reaching into random parts of the FS in such a magical way. I'd prefer if we just didn't have a default instead and fall back to the old behaviour?

Reply:
I was going for mirroring the behaviour of Sytest which assumes a `../synapse` directory. But I suppose it would be:

1. Handy to have the functionality of automatically downloading Complement
2. Best to avoid accidentally running code in `../complement`, in case it doesn't contain Complement

Reply:
Ah, didn't realise that sytest did the same. Still, its a bit toooo magical for my liking if we have a better way of doing it.

Reply:
We now download by default!